### PR TITLE
Fundamental features

### PR DESCRIPTION
**Pull request by latypova-alina**
-Create .ruby-version file
-Add rubocop gem and change files according to its warnings
-Change bin/setup and bin/update files

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  ConsistentQuotesInMultiline: true
+
+Style/AlignParameters:
+  EnforcedStyle: with_fixed_indentation
+
+Metrics/LineLength:
+  Max: 120
+
+Metrics/BlockLength:
+  Enabled: false
```
Comment (vladimirbazhanov) [inline .rubocop.yml]: May be it's better to increase Max size of block, for example, to 32 lines, but not to disable it all?

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+Style/FrozenStringLiteralComment:
+  Enabled: false
+
+Style/Documentation:
+  Enabled: false
```
Comment (vladimirbazhanov) [inline .rubocop.yml]: Pls setup your editor to automatically add newline at the end of file, this is general rule of all unix OSes, all files must have newline at the end. Sublime Text has config option for this, pls find it and set

File: Gemfile [hunk 3]
```diff
@@ -1,29 +1,32 @@
...
-gem 'rails', '~> 5.1.2'
-gem 'pg', '~> 0.18'
-gem 'puma', '~> 3.7'
-gem 'sass-rails', '~> 5.0'
-gem 'uglifier', '>= 1.3.0'
-
-gem 'coffee-rails', '~> 4.2'
-gem 'jbuilder', '~> 2.5'
+gem "coffee-rails", "~> 4.2"
+gem "jbuilder", "~> 2.5"
```
Comment (vladimirbazhanov) [inline Gemfile]: Pls remove jbuilder gem, we never will use it

File: bin/setup [hunk 4]
```diff
@@ -6,33 +6,18 @@ include FileUtils
 # path to your application root.
 APP_ROOT = Pathname.new File.expand_path('../../', __FILE__)
 
-def system!(*args)
```
Comment (vladimirbazhanov) [inline bin/setup]: previouly this file was a "ruby script", so it started with  #!/usr/bin/env ruby magic comment that tells to OS how to execute it. Now you made it "shell script" that will be executed by shell (sh or bash or zsh), so pls correct first line to indicate that it is shell sript

File: bin/setup [hunk 5]
```diff
@@ -6,33 +6,18 @@ include FileUtils
...
 
-  puts '== Installing dependencies =='
-  system! 'gem install bundler --conservative'
-  system('bundle check') || system!('bundle install')
+bundle check --path .bundle > /dev/null 2>&1 ||
+bundle install --path=.bundle $BUNDLER_ARGS
 
-  # Install JavaScript dependencies if using Yarn
-  # system('bin/yarn')
+puts "\n== Preparing database =="
```
Comment (vladimirbazhanov) [inline bin/setup]: this is ruby command, now this file is shell script, pls leave only shell commands here

File: .rubocop.yml (hunk 2 above)
Comment (latypova-alina) [inline .rubocop.yml]: Done

File: Gemfile (hunk 3 above)
Comment (latypova-alina) [inline Gemfile]: Done

File: .rubocop.yml (hunk 1 above)
Comment (latypova-alina) [inline .rubocop.yml]: Maybe, but I saw it was disabled in Rails Base so I thought that was on purpose. But I changed it now

File: .ruby-version [hunk 6]
```diff
@@ -0,0 +1 @@
+2.3.1
```
Comment (vladimirbazhanov) [inline .ruby-version]: 2.3.4

File: spec/spec_helper.rb [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    # `true` in RSpec 4.
+    mocks.verify_partial_doubles = true
+  end
+
+  # This option will default to `:apply_to_host_groups` in RSpec 4 (and will
+  # have no way to turn it off -- the option exists only for backwards
+  # compatibility in RSpec 3). It causes shared context metadata to be
+  # inherited by the metadata hash of host groups and examples, rather than
+  # triggering implicit auto-inclusion in groups with matching metadata.
+  config.shared_context_metadata_behavior = :apply_to_host_groups
```
Comment (vladimirbazhanov) [inline spec/spec_helper.rb]: pls remove all comments from here

File: bin/setup [hunk 8]
```diff
@@ -1,38 +1,21 @@
...
-  # This script is a starting point to setup your application.
-  # Add necessary setup steps to this file.
+# Set up Ruby dependencies via Bundler into .bundle folder
+rm -f .bundle/config
 
-  puts '== Installing dependencies =='
-  system! 'gem install bundler --conservative'
-  system('bundle check') || system!('bundle install')
+bundle check --path .bundle > /dev/null 2>&1 ||
+  bundle install --path=.bundle $BUNDLER_ARGS
```
Comment (vladimirbazhanov) [inline bin/setup]: Add line that creates this BUNDLER_ARGS

File: bin/update [hunk 9]
```diff
@@ -1,29 +1,19 @@
...
-  puts '== Installing dependencies =='
-  system! 'gem install bundler --conservative'
-  system('bundle check') || system!('bundle install')
+bundle check --path .bundle > /dev/null 2>&1 ||
+  bundle install --path=.bundle $BUNDLER_ARGS
 
-  puts "\n== Updating database =="
-  system! 'bin/rails db:migrate'
+# Set up configurable environment variables
+if [ ! -f .env ]; then
```
Comment (vladimirbazhanov) [inline bin/update]: this can be removed

File: bin/update [hunk 10]
```diff
@@ -1,29 +1,19 @@
...
+  cp .env.example .env
+fi
 
-  puts "\n== Removing old logs and tempfiles =="
-  system! 'bin/rails log:clear tmp:clear'
-
-  puts "\n== Restarting application server =="
-  system! 'bin/rails restart'
-end
+bin/rails log:clear tmp:clear
```
Comment (vladimirbazhanov) [inline bin/update]: add rake db:migrate

**Pull request closed without merging**